elastic/apm-server
# Fix updatecli update-beats pipeline

## Motivation/summary

The updatecli pipeline "update-beats" is currently failing with the error message:

```
ERROR: something went wrong in target "beats" : "no changed file to commit"
```

Source: https://github.com/elastic/apm-server/actions/runs/4344796011

This happens when a PR already exists.

This change skips the target execution if a PR already exists and its value is already up-to-date.



## How to test these changes



Execute the following command multiple times:

```shell
BRANCH_NAME=7.17 GIT_USER=ok GIT_EMAIL=ok@ok.com GITHUB_TOKEN=<your_personal_access_token> updatecli -c .ci/update-beats.yml apply
```

it should not fail anymore.

I personally tested it with the following diff on my fork:

```diff
scms:
  default:
    kind: github
    spec:
      user: '{{ requiredEnv "GIT_USER" }}'
      email: '{{ requiredEnv "GIT_EMAIL" }}'
-     owner: elastic
+     owner: reakaleek 
      repository: apm-server
      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
      username: '{{ requiredEnv "GIT_USER" }}'
      branch: '{{ requiredEnv "BRANCH_NAME" }}'
```
